src/parsimmon.js

```diff
@@ -53,9 +53,22 @@ Parsimmon.Parser = (function() {
     }
   }
 
+  // For ensuring we have the right argument types
   function assertParser(p) {
     if (!(p instanceof Parser)) throw new Error('not a parser: '+p);
   }
+  function assertNumber(x) {
+    if (!(typeof x === 'number')) throw new Error('not a number: '+x);
+  }
+  function assertRegex(x) {
+    if (!(x instanceof RegExp)) throw new Error('not a regex: '+x);
+  }
```

Comment:
This should probably say "regexp" in the message to be consistent with JS naming.